parity-crypto/src/publickey/ecdh.rs

```diff
@@ -16,21 +16,43 @@
 
 //! ECDH key agreement scheme implemented as a free function.
 
-use super::{Error, Public, Secret, SECP256K1};
+use super::{Error, Public, Secret};
 use secp256k1::{self, ecdh, key};
 
 /// Agree on a shared secret
 pub fn agree(secret: &Secret, public: &Public) -> Result<Secret, Error> {
-	let context = &SECP256K1;
 	let pdata = {
 		let mut temp = [4u8; 65];
 		(&mut temp[1..65]).copy_from_slice(&public[0..64]);
 		temp
 	};
 
-	let publ = key::PublicKey::from_slice(context, &pdata)?;
-	let sec = key::SecretKey::from_slice(context, secret.as_bytes())?;
-	let shared = ecdh::SharedSecret::new_raw(context, &publ, &sec);
+	let publ = key::PublicKey::from_slice(&pdata)?;
+	let sec = key::SecretKey::from_slice(secret.as_bytes())?;
+	// todo[dvdplm]: could go with new_with_hash_no_panic here (unsafe), but without a benchmark I don't see why.
+	let shared = ecdh::SharedSecret::new_with_hash(&publ, &sec, |x, _| {
```

Comment:
We could go with `new_with_hash_no_panic` here (unsafe), but without a benchmark I don't see why.